fc/code_generation.py

```diff
@@ -195,42 +191,68 @@ def derivative_name(deriv):
-            'var_name': variable_name(variable),
-            'initial_value': model.get_value(variable),
-        })
-        parameter_variables[variable] = i
+    todo_use_qualified_names = set()    # TODO: Remove this. See above.
+    for pvar in protocol_variables:
+        if pvar.is_input and pvar.model_variable is not None:
+            eq = model.get_definition(pvar.model_variable)
+            if eq is not None and not eq.lhs.is_Derivative and len(eq.rhs.atoms(VariableDummy)) == 0:
+
+                # TODO: Remove this. See above.
+                if pvar.short_name in todo_use_qualified_names:
+                    raise NotImplementedError('Need to convert parameter maps to use qualified instead of local names.')
+                todo_use_qualified_names.add(pvar.short_name)
+
+                i = len(parameter_info)
+                parameter_info.append({
+                    'index': i,
+                    'local_name': pvar.short_name,
+                    'var_name': variable_name(pvar.model_variable),
+                    'initial_value': model.get_value(pvar.model_variable),
+                })
+                parameter_variables[pvar.model_variable] = i
 
     # Create output information dicts
     # Each output is associated either with a variable or a list thereof.
     output_info = []
     output_variables = set()
-    for i, output in enumerate(outputs):
-        term = output.rdf_term
-        variables = get_variables_transitively(model, term)
-        output_variables.update(variables)
-        if len(variables) == 0:
-            raise ValueError('No variable annotated as {} found'.format(term))
-        elif len(variables) == 1:
-            length = None  # Not a vector output
-            var_name = variable_name(variables[0])
-        else:
+    todo_use_qualified_names = set()    # TODO: Remove this. See above.
+    for pvar in protocol_variables:
+        if not pvar.is_output:
+            continue
+        elif pvar.model_variable is not None:
+            # Single variable output
+            var_name = variable_name(pvar.model_variable)
+            length = None
+            output_variables.add(pvar.model_variable)
+        elif pvar.vector_variables:
             # Vector output
-            if term in vector_orderings:
-                order = vector_orderings[term]
+            variables = list(pvar.vector_variables)
+            rdf_term = pvar.output_terms[0]
+            if rdf_term in vector_orderings:
+                order = vector_orderings[rdf_term]
                 variables.sort(key=lambda s: order[s.rdf_identity])
+            else:
+                pass
+                # TODO: Add some default ordering for consistent output?
             length = len(variables)
             var_name = [{'index': i, 'var_name': variable_name(s)} for i, s in enumerate(variables)]
+            output_variables.update(variables)
+        else:
+            continue
```

Comment:
There's a vector_ordering for the states, but I don't see where any of the other vector outputs define an order?

Reply:
At present they don't, but the logic is there in `code_generation.py` so it can be supported when needed.

Reply:
I'm updating it now to store a vector_ordering for all. But also the `variables` argument above is defined as `variables = list(pvar.vector_variables)` so maybe we can get rid of `vector_orderings` and just make sure the list `vector_variables` has the order we want?

Reply:
(It definitely works, I'm just asking if there's any reason we need to stick to `vector_orderings`)

Reply:
We needed it originally because code_generation was calling get_transitive... which didn't guarantee ordering. If we're now passing it a list anyway then yes we can just order the list.